### ISSUE TITLE
Broken SASS implementation in case of imports from node_modules.

### ISSUE THREAD

**Issue by StfBauer** [SharePoint/sp-dev-docs#7951]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [X] other (enter in the "Additional environment details" area below)

### Additional environment details

- SPFx 1.4
- Node.js 14.19

### Describe the bug / error

When importing a SASS file from the local node_modules folder, the file gets ignored.

```scss
@import 'node_modules/azure-maps-control/dist/atlas.min.css';
```

Overall, the file won't end up in the lib folder, and simply nothing happens.

What works is the new way of importing SASS files using.

```scss
@use 'node_modules/azure-maps-control/dist/atlas.min.css';
```

Not the file gets included. It cannot be used with CSS Modules because the file will get manipulated with the hash at the end of each class.

The workaround is to create a new file in the web part folder without the `.modules` extension and an underscore upfront ([SASS partial](https://sass-lang.com/guide#topic-4)).

- File: _libraryimport.scss
```scss - 
@use 'node_modules/azure-maps-control/dist/atlas.min.css';
```
in the web part file, this library can then be integrated using the following import:

- File: webpart.module.scss
```scss
.webpartClass{
    :global {
        @import 'libraryimport';
    }
}
```

`:global` because it should be avoided to replace any CSS class declaration of 3rd party libraries because they won't work anymore.

That node_module files get ignored by SASS might be a bug. The longer question is how long CSS Module and SASS will work together in this way because at the moment, they countereffect each other through the latest changes.

If it works like this as a workaround, that is also fine, but then this should get documented somewhere because another developer might end up in the same trap. Not sure what the solution would look like because it opened the following scenarios.

* Abandoned CSS Module
The negative side effect is that the style won't be scoped to the web part only. It can be done today already, but we end up with bleeding styles. One solution would be to have all CSS classes always preprocessed through a unique class for the web.

* Abandoned SCSS
still used the most widely used CSS tool, which would limit the functionalities of SPFx in general. It would make it even more complicated to share Styles across different environments, and it also would mean abandoning a complete web platform.

In the long run, something needs to happen here because both together might not work in future.

One could argue that the files could be loaded like this:

```ts
SPComponentLoader.loadCss('https://atlas.microsoft.com/sdk/javascript/mapcontrol/2/atlas.min.css');
```

Yes, true, but I need to ensure that the JS/TS code matches the SCSS in the file and this only works correctly using the node module for ts/js and CSS from the same source.

### Steps to reproduce

1. Create an SPFx webpart project
2. Install
```sh
npm install --save-dev azure-maps-control
```
3. Import SCSS from the Node Modules folder

```scss
@import 'node_modules/azure-maps-control/dist/atlas.min.css';
```

Check compiled files in the lib folder for the existence of the map CSS.

### Expected behavior

As described already above. It's more an architectural documentation issue.

**Comment by AJIXuMuK** [comment]
@StfBauer - does `@import` work if you `@import ~azure-maps-control` instead of `node_modules` path?

**Comment by StfBauer** [comment]
Nope:

```scss
@import ~azure-maps-control;
```

Causes the following issue:

![image](https://user-images.githubusercontent.com/5503835/168061018-7a30a475-7180-4e34-96f4-cb591333da0a.png)

and

```scss
@import '~azure-maps-control';
```

and the error:

![image](https://user-images.githubusercontent.com/5503835/168061574-cffcaf22-59ab-461f-845b-7aa18fd4b58b.png)

The azure-map-contol is also net @microsoft prefixed.

**Comment by AJIXuMuK** [comment]
@StfBauer 
`@import '~azure-maps-control/dist/atlas.min';` works for me, with all the classes will be prefixed if imported in the module.
`@import './node_modules/azure-maps-control/dist/atlas.min.css';` imports the classes without prefixing them.

**Comment by StfBauer** [comment]
🤦  Could be my problem lt time I checked, it hasn't worked for me.

**Comment by AJIXuMuK** [comment]
@StfBauer - could you please check on your end and close the issue if everything is working correctly?

**Comment by StfBauer** [comment]
@AJIXuMuK, yeah my bad works. Since when can you address the node_modules ~ upfront - I guess no one really is aware of that nor is this hack common and more SPFx specific.

**Comment by AJIXuMuK** [comment]
@StfBauer honestly, I don't know since when exactly. But it seems it was there forever :)

https://docs.microsoft.com/en-us/sharepoint/dev/spfx/office-ui-fabric-integration

**Comment by StfBauer** [comment]
Yeah. but it was never stated anywhere for all node_modules -, I guess this was in since version 1.0 + looks awkward because it seems like a slash is missing.

So in this:

```scss
~@microsoft/sp-office-ui-fabric-core/dist/sass/SPFabricCore.scss
```


Seems like a special ~@microsoft token.
